internal/util/util.go

```diff
@@ -320,6 +320,25 @@ func RunePos(b []byte, i int) int {
 	return CharacterCount(b[:i])
 }
 
+// IndexAnyUnquoted returns the first position is s of a character from chars.
+// Escaped (with '\\') and quoted (with single or double quotes) characters
+// are ignored. Returns -1 if not successful
+func IndexAnyUnquoted(s, chars string) int {
+	var q byte = 0
+	for i := 0; i < len(s); i++ {
```

Comment:
This iterates over raw bytes, not Unicode runes. Although the end result is probably gonna be correct (as long as `chars` includes ASCII characters only) thanks to the fact that UTF-8 is backward compatible with ASCII, it's cleaner to iterate over runes:

```
for _, c := range s {
```

(See e.g. `util.IsSpaces()`, `util.CharacterCountInString()` etc.)

Reply:
The nice thing about the current loop is that I can skip escaped characters simply by saying `i++`. Would that still be valid when I said `for i, c := range s`? (It seems to work in practice.)

Reply:
I don't think so. With a `range` clause, `i` does not control the loop anymore, it is reset to the index of the next element on each iteration, regardless of how we modified it in the previous iteration. https://go.dev/play/p/blyH4nRvXge

You could set `q = '\\'` and then handle it accordingly?

Reply:
Just saw your comment after my force-push. Will use `q` instead of the new `e`.